articles/javascript-dates-manipulation-with-date-fns/index.md

```diff
@@ -0,0 +1,475 @@
+
+Hour (1-12)
+- h – 1, 2, 3, 4, ..., 11 and 12
+- ho – 1st, 2nd, 3rd, 4th, ..., 11th and 12th
+- hh – 01, 02, 03, 04, ..., 11 and 12
+
+Minute
+- m - 0, 1, 2, 4, ..., 54, 55, 55, 57, 58 and 59
+- mo –0th 1st, 2nd, 3rd, 4th, ..., 12th, 58th,59th
+- mm - 00, 01, 02, 04, ..., 54, 55, 56, 57, 58 and 59
+
+Second
+- s – 0, 1, 2, 3, 4, ..., 54, 55, 56, 57, 58 and 59
+- so –0th 1st, 2nd, 3rd, 4th, ..., 12th, 58th, 59th
+- ss – 00, 01, 02, 03, 04, ..., 54, 55, 55, 56, 57, 58 and 5
+
+[Check more Unicode date format for Date-fns](https://date-fns.org/v2.16.1/docs/Unicode-Tokens)
+
+### Parsing and Displaying Date
+As we have explained, date-fns is a collection of many small functions. Use `require('date-fn');` to get started. To start [parsing and displaying](https://date-fns.org/v2.16.1/docs/parse) any date you need to import the functions you require, thus you don't have to import the whole API (work with only what you need). Let's display a simple today's date.
+```js
+//import the funtion you want to use
+const {format} = require('date-fns');
+```
+```js
+const {format} = require('date-fns');
+//today's date
+const today =format(new Date(),'dd.MM.yyyy');
+console.log(today);
+```
+This will display the default current date with the date-fns default format. Now, this is where date format comes in to display the date token to a more human-readable and look exactly the way you want (return the date parsed from string using the given format string).
+
+### Displaying Formatted Date
+<iframe height="400px" width="100%" src="https://repl.it/@kimkimani/date-fns-format?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+Alternatively, you can parse a default date value such as:
+
+```js
+const {format} = require('date-fns');
+//parse a default date value
+const date = new Date('2020/09/19');
+console.log(date);
+console.log(`${format(date, 'dd.MM.yyyy')}`);
+console.log(`${format(date, 'yyyy-MM-dd').toString()}`);
+console.log(`${format(date, 'EEEE, MMMM yyyy')}`);
+console.log(`${format(date, 'EEEE,MMMM do, yyyy hh:mm a')}`);
+console.log(`${format(date, 'MMMM, yyyy')}`);
+console.log(`${format(date, 'MMMM.do.')}`);
+console.log(`${format(date, 'EEEE do HH:mm ')}`);
+console.log(`${format(date, 'EEEE,MMMM do, yyyy ppppp')}`);
+console.log(`${format(date, 'do  MMMM yyyy OOOO')}`);
+```
+The above examples will display the date parsed with several date formats. [Check out more date formats](https://date-fns.org/v2.16.1/docs/format) you can play with and get more ideas of how date-fns formats works.
+
+**Note:** we have only imported the `format` function as it is what we need for the moment, that's one of the dynamic features of date-fns. When formatting date-fns token values try to avoid some [common mistakes](https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md) such as:
```

Comment:
I would suggest you give a brief description of some of the common mistakes and how one can avoid them. For example how one can confuse the `dd` and `DD`.